src/figuresClasses.ts

```diff
@@ -1,11 +1,82 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - You don't need to manually assign the constructor parameters to class properties if you use an access modifier. TypeScript will do that automatically. Consider removing the manual assignments to keep your code cleaner.